assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+    }
+    return $A;
+}
+    
+
+//-------------------------------------------------Complex division-----------------------------------------
+// Function: cx_div(num, [roundto=12])
+// Returns the quotient of two complex numbers in standard form as a paired values in an array. 
+// 
+// Parameters:
+// num: An array of real and imaginary parts of two complex numbers (dividend, divisor) given as paired values
+//      in square brackets. For example, to divide 2-3i by sqrt(2)-i, num = array([2,-3], [sqrt(2),-1]). 
+// roundto: Optional - number of decimal places to which values should be rounded off; 
+//          default is 12 decimal places.
+// 
+// Returns:
+// The quotient of two complex numbers in standard form as a paired values in an array.
+
+function cx_div(array $num, int $roundto=12) {
+    if (count($num)!=2) { echo "cx_Div expects 2 complex numbers"; return "";}
+    $rt=1;
+    $tht=0;
+    $N=array($num[0]);
+    $D=array($num[1]);
+    $D_conj=cx_conj($D);
+    
+    $rn= cx_modul($N);
+    $thn= cx_arg($N);
+    $rconj= cx_modul($D_conj);
+    $thconj= cx_arg($D_conj);
+    
+    $den=($D[0][0])**2 +($D[0][1])**2;
+    
+    $r=$rn*$rconj/$den;
+    $tht=$thn+$thconj;
+    $re=$r*cos($tht);
+    $im=$r*sin($tht);  
+        
+    $st= array([round($re,$roundto) , round($im,$roundto)]);     
+        
+    return $st;
+}
+
+//-------------------------------------Quadratic real and complex roots--------------------------------------
+// Function: cx_quadRoot(a,b,c, [roundto = 12])
+// Returns an array of roots of the quadratic equation f(x) = ax^2 + bx + c. Real roots are returned
+// as an array([r1],[r2]) ordered from the smallest to largest, and the complex roots are returned 
+// as an array([Re1,Im1], [Re2,Im2]). 
+// 
+// Parameters:
+// a: The numerical coefficient of x^2
+// b: The numerical coefficient of x    
+// c: The constant
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 12 decimal places. 
+//
+// Returns:
+// An array of roots (either real or complex) of the quadratic equation.
+
+function cx_quadRoot(float $a, float $b, float $c, int $roundto = 12){
+    $d=$b**2 - 4*$a*$c;
+    if ($d<0){
+
+        $re= round(-$b/(2*$a),$roundto);
+        $im= round(sqrt(abs($d))/(2*$a),$roundto);
+        $im2=-$im;
+        $st=array([$re,$im],[$re,$im2]);
+    }
+        else {
+            $r1=round(((-$b-sqrt($d))/(2*$a)),$roundto);
+            $r2=round(((-$b+sqrt($d))/(2*$a)),$roundto);
+            $st=array([$r1],[$r2]);
+        }
+
+    return $st;
+
+}
+
+//------------------------------------------Formating to standard form-------------------------------------
+// Function: cx_format2std(num, [roundto=3])
+// Returns a string form of a complex number in standard form: a + b i. 
+// 
+// Parameters:
+// num: An array of real and imaginary parts of complex numbers given in square brackets: num = array([Re, Im]).
```

Comment:
Ahh, I see now why you were doing everything as arrays of values, to match the input of this function.  But I think having arrays when not really necessary will be confusing.  I think it'd be nicer to modify this function to accept either a single complex number or an array of complex numbers.  

To handle that, you could add to the start of the function
```
if (!is_array($num[0])) {
  $num = [$num];
}
```
That way if the first entry of $num isn't an array (meaning it's a single complex number) then we'll wrap it in an array so the rest of the code logic will work.